src/python/pants/bin/local_pants_runner.py

```diff
@@ -286,22 +283,27 @@ def _compute_final_exit_code(*codes):
     return max_code
 
   def _run(self):
-    try:
-      self._maybe_handle_help()
+    self._set_start_time(time.time())
 
-      engine_result = self._maybe_run_v2()
-      goal_runner_result = self._maybe_run_v1()
-    finally:
+    # wrap the outer exiter
+    local_exiter = LocalExiter(self._run_tracker, self._repro, exiter=self._exiter)
+    with ExceptionSink.exiter_as(local_exiter):
       try:
-        run_tracker_result = self._run_tracker.end()
-      except ValueError as e:
-        # Calling .end() sometimes writes to a closed file, so we return a dummy result here.
-        logger.exception(e)
-        run_tracker_result = PANTS_SUCCEEDED_EXIT_CODE
-
-    final_exit_code = self._compute_final_exit_code(
-      engine_result,
-      goal_runner_result,
-      run_tracker_result
-    )
-    self._exiter.exit(final_exit_code)
+        self._maybe_handle_help()
```

Comment:
Note that this will `exit()`, so if you're returning in this function, it might be worth to return here as well.